test/integration/persistence_test.go

```diff
@@ -46,24 +46,14 @@ func TestPersistence(t *testing.T) {
 		t.Fatalf("Error creating test pod: %s", err)
 	}
 
-	p := &api.Pod{}
-	for p.Status.Phase != "Running" {
-		p = kubectlRunner.GetPod(podName, podNamespace)
-		time.Sleep(time.Second)
-	}
-
-	// Now restart minikube and make sure the pod is still there.
-	minikubeRunner.RunCommand("stop", true)
-	minikubeRunner.CheckStatus("Stopped")
-
-	minikubeRunner.RunCommand("start", true)
-	minikubeRunner.CheckStatus("Running")
-
 	checkPod := func() error {
-		_, err := kubectlRunner.RunCommand(
-			[]string{"get", "pod", podName, "--namespace", podNamespace})
-		return err
+		p := kubectlRunner.GetPod(podName, podNamespace)
+		if p.Status.Phase != "Running" {
```

Comment:
Should this wait for `Ready` status rather than `Running` phase? IIRC it is the `Ready` status that means the pod is now a valid service endpoint & hence accessible by the service IP.